### ISSUE TITLE
Addition to cluster autoscaler privileges

### ISSUE THREAD

**Issue by SayakMukhopadhyay** [aws/aws-eks-best-practices#322]
**Describe the problem**
I am trying to setup my first EKS cluster and I am going through multiple docs including this one. This is when I noticed a discrepancy between this documentation and the cluster autoscaler one. For the IAM roles that needs to be created, the [cluster autoscaler](https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/cloudprovider/aws/README.md#iam-policy) mentions the following:

```json
{
  "Version": "2012-10-17",
  "Statement": [
    {
      "Effect": "Allow",
      "Action": [
        "autoscaling:DescribeAutoScalingGroups",
        "autoscaling:DescribeAutoScalingInstances",
        "autoscaling:DescribeLaunchConfigurations",
        "autoscaling:DescribeScalingActivities",
        "autoscaling:DescribeTags",
        "ec2:DescribeInstanceTypes",
        "ec2:DescribeLaunchTemplateVersions"
      ],
      "Resource": ["*"]
    },
    {
      "Effect": "Allow",
      "Action": [
        "autoscaling:SetDesiredCapacity",
        "autoscaling:TerminateInstanceInAutoScalingGroup",
        "ec2:DescribeImages",
        "ec2:GetInstanceTypesFromInstanceRequirements",
        "eks:DescribeNodegroup"
      ],
      "Resource": ["*"]
    }
  ]
}
```
whereas the documentation here mentions:
```json
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Effect": "Allow",
            "Action": [
                "autoscaling:SetDesiredCapacity",
                "autoscaling:TerminateInstanceInAutoScalingGroup"
            ],
            "Resource": "*",
            "Condition": {
                "StringEquals": {
                    "aws:ResourceTag/k8s.io/cluster-autoscaler/enabled": "true",
                    "aws:ResourceTag/k8s.io/cluster-autoscaler/<my-cluster>": "owned"
                }
            }
        },
        {
            "Effect": "Allow",
            "Action": [
                "autoscaling:DescribeAutoScalingInstances",
                "autoscaling:DescribeAutoScalingGroups",
                "autoscaling:DescribeScalingActivities",
                "ec2:DescribeLaunchTemplateVersions",
                "autoscaling:DescribeTags",
                "autoscaling:DescribeLaunchConfigurations",
                "ec2:DescribeInstanceTypes"
            ],
            "Resource": "*"
        }
    ]
}
```
This documentation doesn't mention the following actions:
`ec2:DescribeImages`,`ec2:GetInstanceTypesFromInstanceRequirements`,`eks:DescribeNodegroup`. These actions were added in the following PRs, https://github.com/kubernetes/autoscaler/pull/4491, https://github.com/kubernetes/autoscaler/pull/4588

I am still learning EKS and as such I am not sure exactly how to use these 3 extra actions along with a `StringEquals`. Some info would be of great help.

**Comment by jicowan** [comment]
Thanks for submitting this issue. It looks like you need to add `eks:DescribeNodegroup` if you're scaling to zero. The other permissions appear to be required when you use resource based rather than instance based scaling. Can you submit a PR for the change in policy @SayakMukhopadhyay?

**Comment by luarx** [comment]
I agree with @SayakMukhopadhyay ! Those actions should be taken into account...

EKS official documentation mentions that `eks:DescribeNodegroup` is used from 1.24 K8s version [here](https://docs.aws.amazon.com/eks/latest/userguide/kubernetes-versions.html)

I tried to add the action here
```
        {
            "Effect": "Allow",
            "Action": [
                "autoscaling:SetDesiredCapacity",
                "autoscaling:TerminateInstanceInAutoScalingGroup",
                "eks:DescribeNodegroup"
            ],
            "Resource": "*",
            "Condition": {
                "StringEquals": {
                    "aws:ResourceTag/k8s.io/cluster-autoscaler/enabled": "true",
                    "aws:ResourceTag/k8s.io/cluster-autoscaler/<my-cluster>": "owned"
                }
            }
        },
```

**But it does not work**, I suspect because of the **Condition**, the autoscaler pod continues saying:
`1 managed_nodegroup_cache.go:133] Failed to query the managed nodegroup test for the cluster test while looking for labels/taints: AccessDeniedException: User: arn:aws:sts::11111111111:assumed-role/AmazonEKSClusterAutoscalerRole/test is not authorized to perform: eks:DescribeNodegroup on resource: arn:aws:eks:eu-central-1:11111111111:nodegroup/eks-safe-devstaging/test/d2c2b98e-322f-9a47-2313-d2c2b98e`

Do you have any idea so that I can still use this best practice? cc. @jicowan

**Comment by jicowan** [comment]
Is the ASG tagged with the k8s.io/cluster-autoscaler/enabled and k8s.io/cluster-autoscaler/<cluster-name> keys and their corresponding values?

**Comment by luarx** [comment]
Yes, it is!

I think that the problem here is that the `eks:DescribeNodegroup` action is related to the Nodegroup and not to ASG.

To confirm this,  `"aws:ResourceTag/k8s.io/cluster-autoscaler/enabled": "true"` and `"aws:ResourceTag/k8s.io/cluster-autoscaler/<my-cluster>": "owned"` **tags were added manually to one Nodegroup and the cluster autoscaler does not complain about that Nodegroup** as it used to do in my previous message.

So the main question here would be, what is the best practise to tackle this case?
